File: digitalmarketplace/src/main/java/nl/fhict/digitalmarketplace/model/product/VideoGame.java
package nl.fhict.digitalmarketplace.model.product;

import com.fasterxml.jackson.annotation.JsonFormat;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import javax.validation.Valid;
import javax.validation.constraints.NotBlank;
import java.time.LocalDate;
import java.util.Date;
import java.util.List;

@Entity
public class VideoGame extends Product{
    @Column(name = "product_releaseDate")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy/MM/dd")
//    @DateTimeFormat(pattern = "yyyy/MM/dd")
    private Date releaseDate;
    @ManyToMany(targetEntity = Genre.class, cascade = CascadeType.DETACH, fetch= FetchType.EAGER)
    @JoinTable(name = "Product_Genres", joinColumns = {@JoinColumn(name = "product_id")}, inverseJoinColumns = {@JoinColumn(name = "genre_id")})
    private List<Genre> genres;

    public VideoGame() {
    }

    public Date getReleaseDate() {
        return releaseDate;
    }

    public void setReleaseDate(@NotBlank Date releaseDate) {
        this.releaseDate = releaseDate;
    }

    public List<Genre> getGenres() {
        return genres;
    }

    public void setGenres(@Valid List<Genre> genres) {
        this.genres = genres;
    }

    @Override
    public String getType() {
        return "videogame";
    }

    @Override
    public String toString() {
        return "VideoGame{" +
                super.toString()+
                "releaseDate='" + releaseDate + '\'' +
                ", genres=" + genres +
                '}';
    }
}

File: bluedevil-clientapp/src/app/common/videogame.ts
import {Product} from "./product";
import {Genre} from "./genre";
import {ProductPlatform} from "./productplatform";

export class Videogame extends Product{
  public releaseDate: string;
  public genres: Genre[];


  constructor(name: string, platform: ProductPlatform, description: string, imageUrl: string, systemReq: string, active: boolean, price: number, releaseDate: string, genres: Genre[]) {
    super(name, platform, description, imageUrl, systemReq, active, price);
    this.releaseDate = releaseDate;
    this.genres = genres;
    this.setType('videogame');
  }

  getType(): string {
    return "videogame";
  }

}
